qiskit_nature/drivers/second_quantization/hdf5d/hdf5driver.py

```diff
@@ -116,17 +119,24 @@ def run(self) -> GroupedSecondQuantizedProperty:
         with h5py.File(hdf5_file, "r") as file:
             if "origin_driver" in file.keys():
                 legacy_hdf5_file = True
-                LOGGER.warning(
-                    "Your HDF5 file contains the legacy QMolecule object! You should consider "
-                    "converting it to the new property framework, see HDF5Driver.convert"
+                warn_deprecated(
+                    "0.4.0",
+                    DeprecatedType.METHOD,
+                    "HDF5Driver.run with legacy HDF5 file",
+                    additional_msg=(
+                        ". Your HDF5 file contains the legacy QMolecule object! You should consider "
+                        "converting it to the new property framework. See also HDF5Driver.convert"
+                    ),
```

Comment:
I guess we switched from a warning/alert to consider changing to more of an impetus to change with the deprecation.

Reply:
I changed from logger, which the user would have to set to warning level to see it, to a deprecation that will be seen once, because the user should ideally convert the hdf5 file to the new format.